AprilSylph/XKit-Rewritten
# Meatballs util: Enable meatball items in blog menus

### Description


This creates `registerBlogMeatballItem`/ `unregisterBlogMeatballItem`, allowing scripts to add meatball items to the meatball menus in blog cards and the blog view header.

- [x] Update this if #908 is merged.

### Testing steps


Unrevert the demo script commit. Use the meatball menus in a blog card, in the modal blog view header, and in the non-modal blog view header and confirm that they work and adapt to the name of the selected blog.
